### PR TITLE
Add variable to allow the tox version to be specified.

### PR DESCRIPTION
**Pull request by samaloney**
So the idea here is you could set a parameter in the `azure-pipline.yml` config e.g. `toxverspec: <0.4` but I'm not really sure what I'm doing here 🤔

### CODE REVIEW

**Review by Cadair — APPROVED** [review]

File: run-tox-env.yml [hunk 1]
```diff
@@ -187,15 +188,14 @@ jobs:
...
+          - script: docker exec -i -w /project ${{ variables.docker_name }} ${{ variables.setarch }} ${{ variables.docker_python_exec }} -m pip install --upgrade "tox ${{ variables.toxverspec }}" ${{ variables.toxdeps }}
             displayName: Install tox
 
           - script: docker exec -i -w /project ${{ variables.docker_name }} ${{ variables.setarch }} ${{ variables.docker_python_exec }} -m tox -e ${{ env_pair.value }} ${{ variables['toxargs'] }} -- ${{ variables['pytest_flag'] }} ${{ variables['posargs'] }}
             displayName: Running tox
 
         - ${{ if not(variables.docker_image) }}:
-
-          - script: python -m pip install --upgrade tox ${{ variables.toxdeps }}
+          - script: python -m pip install "tox${{ variables.toxverspec }}" ${{ variables.toxdeps }}
```
Comment (Cadair) [inline run-tox-env.yml:198]:
```suggestion

          - script: python -m pip install "tox${{ variables.toxverspec }}" ${{ variables.toxdeps }}
```

**Comment by bsipocz** [comment]
Could you add a test for this in a new PR? That would help downstream to adopt the usage more than than the docs. Thanks!

**Comment by samaloney** [comment]
Yea I guess just add another stanza to the `azure-pipelines.yml` like and test it does what it's supposed to?

https://github.com/OpenAstronomy/azure-pipelines-templates/blob/deefa6efef819d63b10a09e41009d7fadd243ed5/azure-pipelines.yml#L38-L42

**Comment by bsipocz** [comment]
I ended up using the feature here, so a test along these lines should work here, too: 
https://github.com/astropy/pytest-filter-subpackage/pull/12/files

**Comment by samaloney** [comment]
I've added a test in a new PR #87

**Pull request merged**